src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java

```diff
@@ -2179,6 +2181,11 @@ public TextFileType getTextFileType()
    }
    
    @Handler
+   void onShowDocumentOutline()
+   {
```

Comment:
I think you should construct the TextEditingTarget with a reference to the
outline widget (or otherwise provide a reference to it after construction).
I recommend this becauseTextEditingTarget gets to participate in our
"dynamic commands" system which will always route the command to the
currently active editor. I think this is preferable to creating other
ad-hoc means of doing the routing.


On Fri, Jun 12, 2015 at 1:45 PM, Kevin Ushey <notifications@github.com>
wrote:

> In
> src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java
> <https://github.com/rstudio/rstudio/pull/436#discussion_r32337541>:
>
> > @@ -2179,6 +2181,11 @@ public TextFileType getTextFileType()
> >     }
> >
> >     @Handler
> > +   void onShowDocumentOutline()
> > +   {
>
> Ah, yes, we do. Currently the TextEditingTarget doesn't 'know' about the
> associated document outline and so I am not sure if the handler should live
> here. Any thoughts? We could:
>
>    - construct the TextEditingTarget with a reference to the outline
>    widget,
>    - attach the handler to the widget itself (or something else a bit
>    higher up),
>    - use events to coordinate
>
> Any thoughts one what might fit best here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rstudio/rstudio/pull/436/files#r32337541>.
>